lib/pages/chat_adaptive_scaffold/chat_adaptive_scaffold.dart

```diff
@@ -29,111 +23,70 @@ class ChatAdaptiveScaffold extends StatefulWidget {
-                          onBack: toggleRightPanel,
-                          jumpToEventId: jumpToEventId,
-                          isInStack: true,
-                        ),
-                    ],
-                  ),
-                ),
-                const WidthPlatformBreakpoint(
-                  begin: breakpoint,
-                ): SlotLayout.from(
-                  key: AppAdaptiveScaffold.breakpointWebAndDesktopKey,
-                  builder: (_) => Padding(
-                    padding: showRightPanel
-                        ? ChatAdaptiveScaffoldStyle.webPaddingRight
-                        : EdgeInsets.zero,
-                    child: ClipRRect(
-                      borderRadius: ChatAdaptiveScaffoldStyle.borderRadius,
-                      child: body!,
-                    ),
-                  ),
-                ),
-              },
-            ),
-            bodyRatio: 0.7,
-            internalAnimations: false,
-            secondaryBody: showRightPanel
-                ? SlotLayout(
-                    config: {
-                      const WidthPlatformBreakpoint(
-                        end: breakpoint,
-                      ): SlotLayout.from(
-                        key: AppAdaptiveScaffold.breakpointMobileKey,
-                        builder: null,
-                      ),
-                      const WidthPlatformBreakpoint(
-                        begin: breakpoint,
-                      ): SlotLayout.from(
-                        key: AppAdaptiveScaffold.breakpointWebAndDesktopKey,
-                        builder: (_) => ClipRRect(
-                          borderRadius: ChatAdaptiveScaffoldStyle.borderRadius,
-                          child: ChatSearch(
-                            roomId: widget.roomId,
-                            onBack: toggleRightPanel,
-                            jumpToEventId: jumpToEventId,
-                            isInStack: false,
-                          ),
-                        ),
-                      ),
-                    },
-                  )
-                : null,
-          ),
-        );
+    return Navigator(
+      initialRoute: type.initialRoute,
+      onGenerateRoute: (settings) {
+        switch (settings.name) {
+          case RightColumnRouteNames.search:
+            return MaterialPageRoute(
+              builder: (_) => ChatSearch(
+                roomId: roomId,
+                onBack: controller.hideRightColumn,
+                jumpToEventId: controller.jumpToEventId,
+                isInStack: isInStack,
+              ),
+            );
+          case RightColumnRouteNames.profileInfo:
+            return MaterialPageRoute(
+              builder: (_) => ProfileInfo(
+                onBack: controller.hideRightColumn,
+                roomId: roomId,
+              ),
+            );
+        }
+        return MaterialPageRoute(builder: (_) => const SizedBox());
```

Comment:
why did you not use default value on your switch case ?

Reply:
fixed